server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -331,13 +333,30 @@ public void childEvent(
     Assert.assertEquals(ImmutableMap.of(dataSource, 100.0), coordinator.getLoadStatus());
     curator.delete().guaranteed().forPath(ZKPaths.makePath(LOADPATH, dataSegment.getIdentifier()));
     // Wait for coordinator thread to run so that replication status is updated
-    while (coordinator.getSegmentAvailability().snapshot().get(dataSource) != 0) {
-      Thread.sleep(50);
-    }
+    waitFor(
+        new Callable<Boolean>()
+        {
+          public Boolean call()
+          {
+            return coordinator.getSegmentAvailability().snapshot().get(dataSource) == 0;
+          }
+        }, 1500, 50
```

Comment:
since we set the timeout on the overall test, there is no reason to have specific timeouts on specific places in the test.